pkg/performanceprofile/controller/performanceprofile/hypershift/components/handler.go

```diff
@@ -261,14 +262,15 @@ func createOrUpdateKubeletConfigConfigMap(ctx context.Context, cli client.Client
 
 // configMapMeta return a ConfigMap that can be used to encapsulate
 // cluster scoped objects within the desired Namespace
-func configMapMeta(name, profileName, namespace, npNamespacedName string) *corev1.ConfigMap {
+func configMapMeta(name, profileName, namespace, npNamespacedName, ecapsulatedObjKind string) *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: namespace,
 			Name:      name,
 			Labels: map[string]string{
-				hypershiftPerformanceProfileNameLabel: profileName,
-				hypershiftNodePoolLabel:               parseNamespacedName(npNamespacedName),
+				hypershiftPerformanceProfileNameLabel:   profileName,
+				hypershiftNodePoolLabel:                 parseNamespacedName(npNamespacedName),
+				hypershift.ConfigMapEncapsulatedKindKey: strings.ToLower(ecapsulatedObjKind),
```

Comment:
labels are public. Too public. Can we store this information elsewhere?

Reply:
answering myself: no, and these labels aren't accessible by users anyway. There is not a clearly better alternative yet.